lib/cpp/src/thrift/server/TThreadedServer.h

```diff
@@ -20,120 +20,86 @@
-                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(ProcessorFactory, TProcessorFactory))
+    : TServer(processorFactory, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(new concurrency::PlatformThreadFactory),
+      stop_(false) {}
 
   template <typename Processor>
   TThreadedServer(const boost::shared_ptr<Processor>& processor,
-                  const boost::shared_ptr<TServerTransport>& serverTransport,
-                  const boost::shared_ptr<TTransportFactory>& transportFactory,
-                  const boost::shared_ptr<TProtocolFactory>& protocolFactory,
-                  const boost::shared_ptr<ThreadFactory>& threadFactory,
-                  THRIFT_OVERLOAD_IF(Processor, TProcessor));
+                  const boost::shared_ptr<transport::TServerTransport>& serverTransport,
+                  const boost::shared_ptr<transport::TTransportFactory>& transportFactory,
+                  const boost::shared_ptr<protocol::TProtocolFactory>& protocolFactory,
+                  const boost::shared_ptr<concurrency::ThreadFactory>& threadFactory,
+                  THRIFT_OVERLOAD_IF(Processor, TProcessor))
+    : TServer(processor, serverTransport, transportFactory, protocolFactory),
+      threadFactory_(threadFactory),
+      stop_(false) {}
 
-  virtual ~TThreadedServer();
+  virtual ~TThreadedServer() {};
 
   virtual void serve();
 
-  void stop() {
+  virtual void stop() {
     stop_ = true;
     serverTransport_->interrupt();
   }
 
 protected:
-  void init();
+  virtual void disposeClient(TConnectedClient *pClient);
 
-  boost::shared_ptr<ThreadFactory> threadFactory_;
+  boost::shared_ptr<concurrency::ThreadFactory> threadFactory_;
```

Comment:
Any objections to adding the following to each class that is typically wrapped in a shared ptr?  We'll take TTransport as an example, instead of:
const boost::shared_ptr<apache::thrift::transport::TTransport>&
it could be
const apache::thrift::transport::TTransport::Ptr&
This is achieved by adding a typedef of
typedef boost::shared_ptr<TTransport> Ptr;
in the TTransport class header.